libnavigation-ui/src/main/java/com/mapbox/navigation/ui/instruction/InstructionView.java

```diff
@@ -1017,16 +1017,20 @@ private void updateDataFromBannerText(@NonNull BannerText primaryBannerText, Ban
   }
 
   private void loadPrimary(BannerText primaryBannerText) {
-    if (!ViewUtils.isLandscape(getContext())) {
-      stepPrimaryText.setMaxLines(2);
-    }
-    stepSecondaryText.setVisibility(GONE);
-    adjustBannerTextVerticalBias(0.5f);
-
     if (ViewUtils.isLandscape(getContext())) {
-      stepPrimaryText.setMaxWidth(instructionLayoutText.getWidth());
+      stepSecondaryText.setVisibility(GONE);
+      adjustBannerTextVerticalBias(0.5f);
+      int maxWidth = instructionLayoutText.getWidth();
+      if (maxWidth > 0) {
+        stepPrimaryText.setMaxWidth(instructionLayoutText.getWidth());
+      }
+      loadTextWith(primaryBannerText, stepPrimaryText);
```

Comment:
If this is called from both branches of the `if else` we can extract that out, right? 👀 L#1032

Same thing happens with `stepSecondaryText.setVisibility(GONE);` L#1021 and L#1030 and `adjustBannerTextVerticalBias(0.5f);` L#1022 and L#1031

Reply:
😃  actually I intentionally put them into separate blocks so we could clearly see what exactly happens in different orientation. same as below one.
but I can revert it.